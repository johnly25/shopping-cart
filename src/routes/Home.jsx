import { Box } from "@chakra-ui/react";
import background from '../assets/homepage_background.jpg'

function Home() {
    return (
        <>
            <Box h='100vh' bgImage={background} bgSize='cover'>
            </Box>
        </>
    )
}

export default Home;