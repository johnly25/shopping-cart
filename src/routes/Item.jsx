import { useOutletContext, useParams, useRouteLoaderData } from "react-router-dom";
import { Box, Image, Card, Grid, Flex, Button, NumberInput, NumberInputField, NumberInputStepper, NumberIncrementStepper, NumberDecrementStepper } from "@chakra-ui/react";
import { useState } from "react";

function Item() {
    const data = useRouteLoaderData("root");
    const [cart, handleAddToCart] = useOutletContext();
    const [quantity, setQuantity] = useState(1);
    const { id } = useParams();
    const item = getItem(id);   

    function getItem(id) {
        return data.find(x => x.id == id);
    }

    function handleChange(value) {
        setQuantity(parseInt(value));
    }

    return (
        <>
            <Flex justify='center' align='center'>
                <Grid gridTemplateColumns='2fr 1fr' padding='5rem'>
                    <Flex justifyContent='center' alignItems='center'>
                        <Card background='#f1f1f1' h='xl' w='xl' justifyContent='center' alignItems='center'>
                            <Image mixBlendMode={'multiply'} src={item.image} objectFit='contain' width='60%' />
                        </Card>
                    </Flex>
                    <Box>
                        <Grid gap={10}>
                            <Grid gridTemplateColumns='1fr min-content min-content' justify-content='space-between' w='100%' >
                                <Box fontWeight='bold'>{item.title}</Box>
                                <Box>{item.price.toFixed(2)}</Box>
                                <Box></Box>
                            </Grid>
                            <Box fontSize='sm'>{item.description}</Box>
                            <NumberInput size='lg' maxW={32} defaultValue={quantity} min={1} step={1} onChange={handleChange}>
                                <NumberInputField />
                                <NumberInputStepper>
                                    <NumberIncrementStepper />
                                    <NumberDecrementStepper />
                                </NumberInputStepper>
                            </NumberInput>
                            <Flex flexDir='column' gap='1rem'>
                                <Button onClick={() => handleAddToCart(data,id,quantity)}>Add to bag</Button>
                                <Button variant='outline'>Add to wishlist</Button>
                            </Flex>
                        </Grid>
                    </Box>
                </Grid>
            </Flex>
        </>
    )
}

export default Item;