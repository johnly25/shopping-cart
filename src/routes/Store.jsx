import { useRouteLoaderData } from "react-router-dom";
import CardList from "../components/CardList/CardList";

function Store() {
    const data = useRouteLoaderData('root');
    return (
        <>
            {<CardList data={data} />}
        </>
    );
}

export default Store;