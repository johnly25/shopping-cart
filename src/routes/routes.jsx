import App from '../App.jsx'
import Store from './Store.jsx'
import Item from './Item.jsx'
import Home from './Home.jsx'
import api from '../data/api2.jsx'

const routes = [
    {
      path: "/",
      element: <App />,
      loader: async () => {
        // let response = await fetch('https://fakestoreapi.com/products')
        let response = await api;
        return response;
      },
      id: 'root'
      ,
      children: [
        {
          path: "/",
          element: <Home />
        },
        {
          path: "store",
          element: <Store />,
        },
        {
          path: 'store/item/:id',
          element: <Item />
        }
      ],
    },
  ]

export default routes;