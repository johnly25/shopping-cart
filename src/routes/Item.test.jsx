import { fireEvent, render, screen } from "@testing-library/react";
import { beforeEach, describe } from "vitest";
import {
  RouterProvider,
  createMemoryRouter,
} from "react-router-dom";
import routes from "./routes.jsx";
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';


describe('Item', () => {
  let router = createMemoryRouter(routes, {
    initialEntries: ["/store/item/1"]
  });

  test('Item snapshot', async () => {
    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
  });

  test('Item add to bag', async () => {
    const user = userEvent.setup();
    const { container } = render(<RouterProvider router={router} />);
    const button = screen.getByRole('button', { name: 'Add to bag' });
    await user.click(button);
    await user.click(button);

  })

});