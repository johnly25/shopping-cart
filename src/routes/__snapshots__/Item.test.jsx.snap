// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Item > Item snapshot 1`] = `
<div>
  <div
    class="css-s92abg"
  >
    <div
      class="css-ep4wr0"
    >
      Center
    </div>
    <div
      class="css-1mr4ysp"
    >
      <div
        class="css-19pkskf"
      >
        <button
          aria-label="Open Menu"
          class="chakra-button css-n1aaeq"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
              fill="currentColor"
            />
          </svg>
        </button>
        <div
          class="chakra-stack css-pa7cn"
        >
          <a
            class="css-0"
            href="/"
          >
            Logo
          </a>
          <nav
            class="chakra-stack css-1y6oau7"
          >
            <a
              class="chakra-link css-hu4u89"
              href="/store"
            >
              Store
            </a>
          </nav>
        </div>
        <div
          class="css-70qvj9"
        >
          <div
            class="css-79elbk"
          >
            <svg
              class="chakra-icon css-hn76xp"
              data-testid="icon"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M241.55,64.74A12,12,0,0,0,232,60H60.23L51.56,28.79A12,12,0,0,0,40,20H20a12,12,0,0,0,0,24H30.88l34.3,123.49a28.09,28.09,0,0,0,27,20.51H191a28.09,28.09,0,0,0,27-20.51l25.63-92.28A12,12,0,0,0,241.55,64.74ZM194.8,161.07A4,4,0,0,1,191,164H92.16a4,4,0,0,1-3.85-2.93L66.9,84H216.21ZM108,220a20,20,0,1,1-20-20A20,20,0,0,1,108,220Zm104,0a20,20,0,1,1-20-20A20,20,0,0,1,212,220Z"
              />
            </svg>
            <div
              class="css-1qmlj5c"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-gmuwbf"
      >
        <div
          class="css-bpwu4k"
        >
          <div
            class="css-1n2mv2k"
          >
            <div
              class="chakra-card css-bm4qdb"
            >
              <img
                class="chakra-image css-166o4kh"
                src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
              />
            </div>
          </div>
          <div
            class="css-0"
          >
            <div
              class="css-t06pii"
            >
              <div
                class="css-1fxm11x"
                justify-content="space-between"
              >
                <div
                  class="css-722v25"
                >
                  Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
                </div>
                <div
                  class="css-0"
                >
                  109.95
                </div>
                <div
                  class="css-0"
                />
              </div>
              <div
                class="css-itvw0n"
              >
                Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
              </div>
              <div
                class="chakra-numberinput css-yffede"
              >
                <input
                  aria-valuemax="9007199254740991"
                  aria-valuemin="1"
                  aria-valuenow="1"
                  aria-valuetext="1"
                  autocomplete="off"
                  autocorrect="off"
                  class="chakra-numberinput__field css-1dqab6m"
                  inputmode="decimal"
                  pattern="[0-9]*(.[0-9]+)?"
                  role="spinbutton"
                  type="text"
                  value="1"
                />
                <div
                  aria-hidden="true"
                  class="css-1jj9yua"
                >
                  <div
                    class="css-1m5jnul"
                    role="button"
                    tabindex="-1"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                  <div
                    aria-disabled="true"
                    class="css-1m5jnul"
                    disabled=""
                    role="button"
                    tabindex="-1"
                  >
                    <svg
                      class="chakra-icon css-onkibi"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="css-19s3t1h"
              >
                <button
                  class="chakra-button css-ez23ye"
                  type="button"
                >
                  Add to bag
                </button>
                <button
                  class="chakra-button css-11494p6"
                  type="button"
                >
                  Add to wishlist
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-1gx57aq"
    >
      <div
        class="chakra-stack chakra-container css-1ye2k0l"
      >
        <div
          class="css-1xk03wa"
        >
          <div
            class="chakra-stack css-1ey9w6j"
          >
            <p
              class="chakra-text css-dculj6"
            >
              Company
            </p>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              About Us
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Blog
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Careers
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Contact Us
            </a>
          </div>
          <div
            class="chakra-stack css-1ey9w6j"
          >
            <p
              class="chakra-text css-dculj6"
            >
              Support
            </p>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Help Center
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Safety Center
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Community Guidelines
            </a>
          </div>
          <div
            class="chakra-stack css-1ey9w6j"
          >
            <p
              class="chakra-text css-dculj6"
            >
              Legal
            </p>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Cookies Policy
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Privacy Policy
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Terms of Service
            </a>
            <a
              class="chakra-link css-spn4bz"
              href="#"
            >
              Law Enforcement
            </a>
          </div>
          <div
            class="chakra-stack css-1ey9w6j"
          >
            <p
              class="chakra-text css-dculj6"
            >
              Install App
            </p>
          </div>
        </div>
      </div>
      <div
        class="css-17jet92"
      >
        <div
          class="chakra-stack chakra-container css-f2r279"
        >
          <p
            class="chakra-text css-0"
          >
            Â© 2022 Chakra Templates. All rights reserved
          </p>
          <div
            class="chakra-stack css-1f6hfsa"
          >
            <a
              class="css-hyb7bs"
              href="#"
            >
              <span
                class="css-idkz9h"
              >
                Twitter
              </span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                />
              </svg>
            </a>
            <a
              class="css-hyb7bs"
              href="#"
            >
              <span
                class="css-idkz9h"
              >
                YouTube
              </span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 576 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
                />
              </svg>
            </a>
            <a
              class="css-hyb7bs"
              href="#"
            >
              <span
                class="css-idkz9h"
              >
                Instagram
              </span>
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
